pkg/service/FluxService/FluxApplicationService.go

```diff
@@ -0,0 +1,420 @@
+	fluxAppTreeResponse := make([]*bean.ResourceTreeResponse, 0)
+
+	k8sConfig := impl.converter.GetClusterConfigFromClientBean(request.Config)
+	restConfig, err := impl.k8sUtil.GetRestConfigByCluster(k8sConfig)
+	resp, err := impl.k8sUtil.GetResource(context.Background(), request.Namespace, request.Name, GvkForHelmreleaseFluxApp, restConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting resource", "err", err)
+		return nil, nil, err
+	}
+
+	releaseName := ""
+	_namespace := ""
+	var appStatus *FluxAppStatusDetail
+
+	if resp != nil && resp.Manifest.Object != nil {
+		releaseName, _namespace, err = getReleaseNameNamespace(resp.Manifest.Object, request.Name)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		appStatus, err = getKsAppStatus(resp.Manifest.Object, GvkForHelmreleaseFluxApp)
+		if err != nil {
+			impl.logger.Errorw("error in getting app status", "err", err)
+			return nil, nil, err
+		}
+
+		if releaseName == "" && _namespace == "" {
+			impl.logger.Errorw("error in getting release name and namespace", "err", err)
+			return nil, appStatus, err
+		}
+	}
+	helmRelease, err := impl.common.GetHelmRelease(request.Config, _namespace, releaseName)
+	if err != nil {
+		impl.logger.Errorw("error in getting helm release", "err", err)
+		return nil, appStatus, err
+	}
+	appDetailRequest := &client.AppDetailRequest{
+		ReleaseName:   releaseName,
+		Namespace:     _namespace,
+		ClusterConfig: request.Config,
+	}
+
+	resourceTreeResponse, err := impl.common.BuildResourceTree(appDetailRequest, helmRelease)
+	if err != nil {
+		impl.logger.Errorw("error in building resource tree", "err", err)
+		return nil, appStatus, err
+	}
+	fluxAppTreeResponse = append(fluxAppTreeResponse, resourceTreeResponse)
+
+	return fluxAppTreeResponse, appStatus, nil
+}
+func (impl *FluxApplicationServiceImpl) buildFluxAppDetailForKustomize(request *FluxAppDetailRequest) ([]*bean.ResourceTreeResponse, *FluxAppStatusDetail, error) {
+	fluxAppTreeResponse := make([]*bean.ResourceTreeResponse, 0)
+
+	k8sClusterConfig := impl.converter.GetClusterConfigFromClientBean(request.Config)
+	restConfig, err := impl.k8sUtil.GetRestConfigByCluster(k8sClusterConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting the rest config", "error", err)
+		return nil, nil, err
+	}
+	resp, err := impl.k8sUtil.GetResource(context.Background(), request.Namespace, request.Name, GvkForKustomizationFluxApp, restConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting app manifest", "err", err)
+		return nil, nil, err
+	}
+	if resp == nil || resp.Manifest.Object == nil {
+		impl.logger.Errorw("app manifest's response or manifest object is nil", "err", err)
+		return nil, nil, err
+	}
+	appStatus, err := getKsAppStatus(resp.Manifest.Object, GvkForKustomizationFluxApp)
+	if err != nil {
+		impl.logger.Errorw("failed to get app status", "err", err)
+		return nil, nil, err
+	}
+	if !inventoryExists(resp.Manifest.Object) {
```

Comment:
inventoryExists(resp.Manifest.Object) -> inventory, ok := !inventoryExists(resp.Manifest.Object); !ok

return inventory if exists from this func , and later instead of passing resp.Manifest.Object in getFluxResourceAndFluxHrList then inside that func, instead of fetching  inventory from manifest again just use that inventory instead